crates/transaction-pool/src/blobstore/disk.rs

```diff
@@ -52,22 +52,34 @@ impl BlobStore for DiskFileBlobStore {
 
     fn insert_all(&self, txs: Vec<(B256, BlobTransactionSidecar)>) -> Result<(), BlobStoreError> {
         if txs.is_empty() {
-            return Ok(())
+            return Ok(());
         }
         self.inner.insert_many(txs)
     }
 
     fn delete(&self, tx: B256) -> Result<(), BlobStoreError> {
-        self.inner.delete_one(tx)?;
+        let mut files_to_delete = self.inner.files_to_delete.write();
+        files_to_delete.insert(tx);
         Ok(())
     }
 
     fn delete_all(&self, txs: Vec<B256>) -> Result<(), BlobStoreError> {
-        if txs.is_empty() {
-            return Ok(())
+        let mut files_to_delete = self.inner.files_to_delete.write();
+        for tx in txs {
+            files_to_delete.insert(tx);
         }
+        Ok(())
+    }
 
-        self.inner.delete_many(txs)?;
+    fn cleanup(&self) -> Result<(), BlobStoreError> {
+        let mut files_to_delete = self.inner.files_to_delete.write();
+        let txs_to_delete = std::mem::take(&mut *files_to_delete);
```

Comment:
we can prob scope this so the `RwLockWriteGuard` is dropped after we do `std::mem::take` to not hold on to the guard too long